### ISSUE TITLE
About convert_coco

### ISSUE THREAD

**Issue by wsy-yjys** [ultralytics/ultralytics#10024]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

```python
from ultralytics.data.converter import convert_coco
convert_coco(labels_dir=r'/data01/wsy/COCO/annotations/',
             save_dir=r'/data01/wsy/COCO/annotations/',
             use_segments=False,
             lvis=True)
```

When I use the above code to convert the coco dataset json annotation to yolo format, is it normal to find 1021 backgrounds in the training set and 48 backgrounds in the validation set when training?

```bash
[34m[1mtrain: [0mScanning /data01/wsy/coco/labels/train2017... 117266 images, 1021 backgrounds, 0 corrupt: 100%|██████████| 118287/118287 [07:59<00:00, 246.82it/s]
[34m[1mtrain: [0mNew cache created: /data01/wsy/coco/labels/train2017.cache

[34m[1mval: [0mScanning /data01/wsy/coco/labels/val2017... 4952 images, 48 backgrounds, 0 corrupt: 100%|██████████| 5000/5000 [00:05<00:00, 845.59it/s]
[34m[1mval: [0mNew cache created: /data01/wsy/coco/labels/val2017.cache
```

### Environment

```bash
(torch2.2) wsy@ubuntu:~/paper2/ultralytics_240412$ yolo checks
/home/wsy/anaconda3/envs/torch2.2/lib/python3.9/site-packages/torch/cuda/__init__.py:141: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11040). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
  return torch._C._cuda_getDeviceCount() > 0
Ultralytics YOLOv8.1.47 🚀 Python-3.9.19 torch-2.2.0+cu121 CPU (Intel Xeon Gold 5218R 2.10GHz)
Setup complete ✅ (80 CPUs, 125.6 GB RAM, 140.3/701.4 GB disk)

OS                  Linux-5.4.0-150-generic-x86_64-with-glibc2.27
Environment         Linux
Python              3.9.19
Install             pip
RAM                 125.56 GB
CPU                 Intel Xeon Gold 5218R 2.10GHz
CUDA                None

matplotlib          ✅ 3.8.4>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.28.1>=2.23.0
scipy               ✅ 1.13.0>=1.4.1
torch               ✅ 2.2.0+cu121>=1.8.0
torchvision         ✅ 0.17.0+cu121>=0.9.0
tqdm                ✅ 4.66.2>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@wsy-yjys yes this is correct. A small percentage of the COCO images in both train and val have no annotations at all. They are "background" images without instances.

**Comment by glenn-jocher** [comment]
@wsy-yjys the COCO dataset does indeed include a small percentage of images that have no labeled objects at all. In these cases, the entire image is essentially considered "background" since no object annotations are provided.

When converting COCO annotations to YOLO format using `ultralytics.data.converter.convert_coco`, images with no labels will result in a "background" class being created in the corresponding YOLO label file. This is expected behavior, as the converter needs to generate a label file for every image, even if no objects are present.

So in your case, finding 1021 "background" labels in the training set and 48 in the validation set is normal and nothing to be concerned about. It simply indicates that those images had no annotated objects in the original COCO dataset.

During YOLO training, these "background" labels will be appropriately handled and should not negatively impact the training process. The model will learn to differentiate between the actual object classes and the background class representing images with no labeled objects.

I apologize for not considering this aspect of the COCO dataset in my initial response. Thank you for bringing it to my attention and allowing me to provide a more accurate explanation.
